roc/log_test.go

```diff
@@ -0,0 +1,100 @@
+package roc
+
+import (
+	"io/ioutil"
+	"log"
+	"testing"
+	"time"
+)
+
+const defaultLogLevel LogLevel = LogError
+
+type testWriter struct {
+	ch chan string
+}
+
+func makeTestWriter() testWriter {
+	return testWriter{
+		ch: make(chan string, 1),
+	}
+}
+
+func (tw testWriter) Write(buf []byte) (int, error) {
+	select {
+	case tw.ch <- string(buf):
```

Comment:
why do you select here? isn't it the same as to do `tw.ch <- string(buf)`? Or you're afraid that it's going to block as nobody is reading it form the other side?

Reply:
To ensure that Write() wont block. Otherwise, if Roc will print logs syncrhonously, OpenContext() will hang when trying to print more than cap(tw.ch) messages, i.e more than one message. We either need a channel with big capacity or we can drop messages if the channel is full. I followed the second approach here.

I've added a comment.

Reply:
all clear now